File: iot-devices/apps/labs/common/DataUtil.py
'''
Created on Feb 18, 2019
toJsonFromSensorData and  toSensorDataFromJson
@author: alex
'''
from labs.common import ActuatorData
from labs.common import SensorData
import json


class DataUtil:
    
    def jsonToActuatorData(self, jsonData):
        
        with open(jsonData, encoding='utf-8') as jsonData:
                adDict = json.loads(jsonData.read())
                
        adDict = json.loads(jsonData)
        print(" decode [pre]  --> " + str(adDict))
        ad = ActuatorData.ActuatorData()
        ad.name = adDict['name']
        ad.timeStamp = adDict['timeStamp']
        ad.hasError = adDict['hasError']
        ad.command = adDict['command']
        ad.errCode = adDict['errCode']
        ad.statusCode = adDict['statusCode']
        ad.stateData = adDict['stateData']
        ad.curValue = adDict['curValue']
        #print(" decode [post] --> " + str(ad))
        return ad
    
    def jsonToSensorData(self, jsonData):
        
        with open(jsonData, encoding='utf-8') as jsonData:
                sdDict = json.loads(jsonData.read())
        
        sdDict = json.loads(jsonData)
        #print(" decode [pre]  --> " + str(sdDict))
        sd = SensorData.SensorData()
        sd.name = sdDict['name']
        sd.timeStamp = sdDict['timeStamp']
        sd.avgValue = sdDict['avgValue']
        sd.minValue = sdDict['minValue']
        sd.maxValue = sdDict['maxValue']
        sd.curValue = sdDict['curValue']
        sd.totValue = sdDict['totValue']
        sd.sampleCount = sdDict['sampleCount']
        
        #print(" decode [post] --> " + str(sd))
        return sd
    
    
    def toJson(self, name,timeStamp,avgValue,minValue,maxValue,curValue,totValue,sampleCount):
            
            json_info = {}
            json_info['name'] = name
            json_info['timeStamp'] = timeStamp
            json_info['avgValue'] = avgValue
            json_info['minValue'] = minValue
            json_info['maxValue'] = maxValue
            json_info['curValue'] = curValue
            json_info['totValue'] = totValue
            json_info['sampleCount'] = sampleCount
            
            with open("/Users/alex/git/iot-devices/data/data.json","w+") as j:
                
                print("Dictionary:  ")
                print(json_info)
                json.dump(json_info,j)
                
                
                
                jsonData = json.dumps(json_info)
                # write data into json file
    
                print("JSon:  ")
                print(jsonData)
            
            return jsonData

File: iot-gateway/src/main/java/NEU/ZuoHao/connecteddevices/labs/common/DataUtil.java
package NEU.ZuoHao.connecteddevices.labs.common;

import java.io.FileReader;

import com.google.gson.Gson;
import com.google.gson.JsonIOException;
import com.google.gson.JsonObject;
import com.google.gson.JsonParser;
import com.google.gson.JsonSyntaxException;

import java.io.FileNotFoundException;

public class DataUtil {
	
	// fromJson method is to read json string from json file
	public SensorData fromJson() throws JsonIOException, JsonSyntaxException, FileNotFoundException {
		//call sensordata method 
		SensorData sensordata = new SensorData();

		JsonObject jo = new JsonParser().parse(new FileReader("/Users/alex/git/iot-devices/data/data.json")).getAsJsonObject();
		// read file by fileReader covert data into SensorData
		sensordata.name = jo.get("name").getAsString();
		sensordata.timeStamp = jo.get("timeStamp").getAsString();
		sensordata.avgValue = jo.get("avgValue").getAsFloat();
		sensordata.minValue = jo.get("minValue").getAsFloat();
		sensordata.maxValue = jo.get("maxValue").getAsFloat();
		sensordata.curValue = jo.get("curValue").getAsFloat();
		sensordata.totValue = jo.get("totValue").getAsFloat();
		sensordata.sampleCount = jo.get("sampleCount").getAsInt();
		//print all of data and check result
		System.out.println("Name: " + sensordata.name);
		System.out.println("Timestamp: " + sensordata.timeStamp);
		System.out.println("Average : " + sensordata.avgValue);
		System.out.println("Min: " + sensordata.minValue);
		System.out.println("Max: " + sensordata.maxValue);
		System.out.println("Current : " + sensordata.curValue);
		System.out.println("totValue : " + sensordata.totValue);
		System.out.println("SampleCount: " + sensordata.sampleCount);
		
		return sensordata;
		
	}
	
	// convert sensorData instance to Json file
	public String sensorDataToJson(SensorData sensorData) {
		String jsonData = null;
		
		if(sensorData != null) {
			Gson gson = new Gson();
			jsonData = gson.toJson(sensorData);
		}
		System.out.println(jsonData);
		
		return jsonData;
	}
	
	// convert jsondata to actuatorData
	public ActuatorData jsonToActuatorData(String jsonData) {
		ActuatorData actuatorData = null;
		
		if(jsonData !=null && jsonData.trim().length() > 0) {
			Gson gson = new Gson();
			actuatorData = gson.fromJson(jsonData, ActuatorData.class);
		}
		
		return actuatorData;
	}

}
